src/executor/graph_executor.cc

```diff
@@ -767,6 +767,28 @@ void GraphExecutor::InitOpSegs() {
   return;
 }
 
+void GraphExecutor::ExecuteMonCallback(size_t nid) {
+  static const auto& flist_outputs =
+      nnvm::Op::GetAttr<nnvm::FListOutputNames>("FListOutputNames");
+  const auto& idx = graph_.indexed_graph();
```

Comment:
This might be slow. Is it doing topo sort every time?
How about input const OpNode& instead